In [3]:
import pandas as pd
import numpy as np

# Problem 1:

Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the Brady kids on The Brady Bunch. Cousin Oliver was 8 years old when he joined the show. What are the mean, median, and mode of the kids' ages when they first appeared on the show? What are the variance, standard deviation, and standard error?

In [11]:
bradyDataFrame = pd.DataFrame()
bradyDataFrame["Names"] = ["Greg", "Marcia", "Peter", "Jan", "Bobby", "Cindy", "Cousin Oliver"]
bradyDataFrame["Ages"] = [14, 12, 11, 10, 8, 6, 8]

print("Mean: " + str(bradyDataFrame["Ages"].mean()))
print("Median: " + str(bradyDataFrame["Ages"].median()))
print("Mode: " + str(bradyDataFrame["Ages"].mode()))

print("Variance: " + str(bradyDataFrame["Ages"].var()))
print("Standard Deviation: " + str(bradyDataFrame["Ages"].std()))
print("Standard Error: " + str(bradyDataFrame["Ages"].sem()))

Mean: 9.857142857142858
Median: 10.0
Mode: 0    8
dtype: int64
Variance: 7.476190476190475
Standard Deviation: 2.734262327610589
Standard Error: 1.0334540197243192


# Problem 2:

Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?

Since the data is symmetrical, I'm going to choose to use the mean as the estimate of the central tendency, and the standard deviation as the estimate of the variance.

# Problem 3:

Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?

In [12]:
bradyDataFrame.loc[bradyDataFrame["Names"] == "Cindy", "Ages"] += 1

print("Mean: " + str(bradyDataFrame["Ages"].mean()))
print("Median: " + str(bradyDataFrame["Ages"].median()))
print("Mode: " + str(bradyDataFrame["Ages"].mode()))

print("Variance: " + str(bradyDataFrame["Ages"].var()))
print("Standard Deviation: " + str(bradyDataFrame["Ages"].std()))
print("Standard Error: " + str(bradyDataFrame["Ages"].sem()))

Mean: 10.0
Median: 10.0
Mode: 0    8
dtype: int64
Variance: 6.333333333333333
Standard Deviation: 2.516611478423583
Standard Error: 0.9511897312113418


All of the variance estimates changed. The mean also changed. The only two estimates that did not change were the median and the mode.

# Problem 4:

Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods?

In [13]:
bradyDataFrame.loc[(bradyDataFrame["Names"] == "Cousin Oliver") & (bradyDataFrame["Ages"] == 8)] = [["Jessica", 1]]

print("Mean: " + str(bradyDataFrame["Ages"].mean()))
print("Median: " + str(bradyDataFrame["Ages"].median()))
print("Mode: " + str(bradyDataFrame["Ages"].mode()))

print("Variance: " + str(bradyDataFrame["Ages"].var()))
print("Standard Deviation: " + str(bradyDataFrame["Ages"].std()))
print("Standard Error: " + str(bradyDataFrame["Ages"].sem()))

Mean: 9.0
Median: 10.0
Mode: 0     1
1     7
2     8
3    10
4    11
5    12
6    14
dtype: int64
Variance: 18.0
Standard Deviation: 4.242640687119285
Standard Error: 1.6035674514745462


Since the data is no longer as symmetrical, I would use median as the estimate of central tendency. I would also use variance to describe the non-normal distribution.

# Problem 5:

On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. The answers were: TV Guide 20% fans Entertainment Weekly 23% fans Pop Culture Today 17% fans SciPhi Phanatic 5% fans

Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

In [14]:
percentageArray = np.array([20, 23, 17, 5])

print("Percentage Median: " + str(np.median(percentageArray)))
print("Percentage Mean: " + str(np.mean(percentageArray)))

Percentage Median: 18.5
Percentage Mean: 16.25


I would use the median to estimate that around 18.5% of adult Americans were fans of the Brady Bunch. I would use median over mean because the data was non-symmetrical. The 5% from a single magazine tanked the mean, while the median was able to have an estimate closer to the middle of the data.